plugins/xkcd.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+import sys
+import json
+import requests
+import random
+
+headers = {
+    'User-Agent': 'Godzilla/5.0 (X11; Linus x86_64; rv:61.0)'
+}
+
+endpoint = "https://xkcd.com/{}/info.0.json"
+
+latest = requests.get(
+    "http://xkcd.com/info.0.json",
+    headers=headers
+).json()['num']
+
+
+def getJSON(id):
+    data = requests.get(
+        endpoint.format(id),
+        headers=headers
+    ).json()
+    return data
+
+
+def response(id, title, url, alt):
+    reply = "#{}: {}\n".format(id, title) + \
+            "{}\n".format(url) + \
+            "*{}*".format(alt)
+    return reply
+
+
+def getComic(id):
+    data = getJSON(id)
+    comic = response(id, data['safe_title'], data['img'], data['alt'])
+    return comic
+
+if len(sys.argv) < 3:
+    reply = getComic(latest)
+else:
+    if sys.argv[2] == "rand":
+        id = random.randint(1, latest)
+        reply = getComic(id)
+    else:
+        try:
+            if int(sys.argv[2]) in range(1, latest):
+                reply = getComic(int(sys.argv[2]))
+        except ValueError:
```

Comment:
not an issue, just wondering what would cause a ValueError here?

Reply:
I think the issue was if the argument was not rand nor a string that can be cast into int then it would give a valueError.